src/Bridge/Doctrine/MongoDB/Extension/PaginationExtension.php

```diff
@@ -0,0 +1,115 @@
+    private $clientItemsPerPage;
+    private $itemsPerPage;
+    private $pageParameterName;
+    private $enabledParameterName;
+    private $itemsPerPageParameterName;
+
+    public function __construct(ManagerRegistry $managerRegistry, RequestStack $requestStack, ResourceMetadataFactoryInterface $resourceMetadataFactory, bool $enabled = true, bool $clientEnabled = false, bool $clientItemsPerPage = false, int $itemsPerPage = 30, string $pageParameterName = 'page', string $enabledParameterName = 'pagination', string $itemsPerPageParameterName = 'itemsPerPage')
+    {
+        $this->managerRegistry = $managerRegistry;
+        $this->requestStack = $requestStack;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->enabled = $enabled;
+        $this->clientEnabled = $clientEnabled;
+        $this->clientItemsPerPage = $clientItemsPerPage;
+        $this->itemsPerPage = $itemsPerPage;
+        $this->pageParameterName = $pageParameterName;
+        $this->enabledParameterName = $enabledParameterName;
+        $this->itemsPerPageParameterName = $itemsPerPageParameterName;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function applyToCollection(Builder $queryBuilder, string $resourceClass, string $operationName = null)
+    {
+        $request = $this->requestStack->getCurrentRequest();
+        if (null === $request) {
+            return;
+        }
+
+        $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
+        if (!$this->isPaginationEnabled($request, $resourceMetadata, $operationName)) {
+            return;
+        }
+
+        $itemsPerPage = $resourceMetadata->getCollectionOperationAttribute($operationName, 'pagination_items_per_page', $this->itemsPerPage, true);
+        if ($resourceMetadata->getCollectionOperationAttribute($operationName, 'pagination_client_items_per_page', $this->clientItemsPerPage, true)) {
+            $itemsPerPage = (int) $request->query->get($this->itemsPerPageParameterName, $itemsPerPage);
+        }
+
+        $queryBuilder
+            ->skip(($request->query->get($this->pageParameterName, 1) - 1) * $itemsPerPage)
+            ->limit($itemsPerPage)
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsResult(string $resourceClass, string $operationName = null) : bool
+    {
+        $request = $this->requestStack->getCurrentRequest();
+        if (null === $request) {
+            return false;
+        }
+
+        $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
+
+        return $this->isPaginationEnabled($request, $resourceMetadata, $operationName);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getResult(Builder $queryBuilder)
+    {
+        return new Paginator($queryBuilder->getQuery()->execute());
+    }
+
+    private function isPaginationEnabled(Request $request, ResourceMetadata $resourceMetadata, string $operationName = null) : bool
```

Comment:
That looks like to be copy/pasted, or at least, that's not specific to MongoDb. Is there a way you can decouple that?